packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/utils/broadcastTx.ts

```diff
@@ -0,0 +1,92 @@
+import { Connection } from "@solana/web3.js"
+import { getConnection } from "./connections"
+import { cacheTransaction, getCachedDiscoveryNodes } from "../redis"
+import { Logger } from "pino"
+import { logger as defaultLogger } from "../logger"
+import { personalSign } from "eth-sig-util"
+import { config } from "../config"
+
+/**
+ * Forwards the transaction response to other Solana Relays on other discovery
+ * nodes so that they can cache it to lighten the RPC load on indexing.
+ */
+export const forwardTransaction = async (logger: Logger, transaction: string) => {
+    const endpoints = await getCachedDiscoveryNodes()
+    logger.info(`Forwarding to ${endpoints.length} endpoints...`)
+    const body = JSON.stringify({ transaction })
+    await Promise.all(
+        endpoints
+            .filter((p) => p.endpoint !== config.endpoint)
+            .map(({ endpoint }) =>
+                fetch(`${endpoint}/solana/cache`, {
+                    method: 'POST',
+                    body,
+                    headers: {
+                        'content-type': 'application/json',
+                        'Discovery-Signature': personalSign(config.delegatePrivateKey, {
+                            data: body
+                        })
+                    }
+                })
+                    .then((res) => {
+                        if (res.ok) {
+                            logger.info(
+                                { endpoint, status: res.status },
+                                `Forwarded successfully`
+                            )
+                        } else {
+                            logger.warn(
+                                { endpoint },
+                                `Failed to forward transaction to endpoint: ${res.statusText}`
+                            )
+                        }
+                    })
+                    .catch((e) => {
+                        logger.warn(
+                            { endpoint },
+                            `Failed to forward transaction to endpoint: ${e}`
+                        )
+                    })
+            )
+    )
+}
+
+export const broadcastTx = async ({ signature, confirm, logger }: { signature: string, confirm: boolean, logger?: Logger }) => {
```

Comment:
one more super nit: can we add a doc comment for this method? It might be unclear what the differences are between this and `forwardTransaction` by name alone